galaxy_importer/main.py

```diff
@@ -98,27 +107,43 @@ def call_importer(args, cfg):  # pragma: no cover
 
     Method excluded from pytest unit test coverage, tests exist in tests/integration
     """
-    if not args.file:
-        return collection.import_collection(
-            git_clone_path=os.path.abspath(args.git_clone_path),
-            output_path=os.path.abspath(args.output_path),
-            logger=logger,
-            cfg=cfg,
-        )
-
-    match = FILENAME_REGEXP.match(os.path.basename(args.file))
-    namespace, name, version = match.groups()
-    filename = collection.CollectionFilename(namespace, name, version)
-
-    with open(args.file, "rb") as fh:
+    if args.legacy_role:
+        if args.file is None:
+            logger.error("Must supply the directory of the role")
+            return None
+        if args.namespace is None:
+            logger.error("Importing legacy role requires explicit namespace")
+            return None
         try:
-            data = collection.import_collection(fh, filename, logger=logger, cfg=cfg)
+            data = legacy_role.import_legacy_role(args.file, args.namespace, logger=logger, cfg=cfg)
         except ImporterError as e:
             logger.error(f"The import failed for the following reason: {str(e)}")
```

Comment:
For collections, the call to schema catches the ValueError and makes it an ImporterError https://github.com/ansible/galaxy-importer/blob/56a7c1a37d5d144a93af94aca82c2df9b18b68e4/galaxy_importer/loaders/collection.py#L223-L226

This allows for expected errors to display cleanly with no traceback for the caller. one possible test you could write here for TDD is that an error like bad dependency should not end in a traceback.

expected:
```
(.env) awc:cisco_ios_test$ python -m galaxy_importer.main awcrosby-cisco_ios_test-1.0.4.0.tar.gz 
Importing with galaxy-importer 0.4.10
ERROR: The import failed for the following reason: Invalid collection metadata. Expecting 'version' to be in semantic version format, instead found '1.0.4.0'.
(.env) awc:cisco_ios_test$
```

legacy role exposes traceback:
```
(.env) awc:ansible-role-jenkins$ python -m galaxy_importer.main --legacy-role . --namespace awcrosby
Importing with galaxy-importer 0.4.10
ERROR: Unexpected error occurred: geerlingguy.java.wrong must have namespace and name separated by '.'
Traceback (most recent call last):
...
ValueError: geerlingguy.java.wrong must have namespace and name separated by '.'
(.env) awc:ansible-role-jenkins$
```